scorers.py

```diff
@@ -255,6 +279,42 @@ def _kendall_tau_distance(self, alignment):
           dis += 1
     return 2*dis/(n*n-n)  
 
+  def cache_stats(self, ref, out):
```

Comment:
This seems like it should work for every scoring function that is decomposable sentence-wise. Maybe we could make a super-class `SentenceFactoredScorer` or something like that? This could then be applied to RIBES, and any other similar measure (e.g. like METEOR if we get it later).